p44/src/hashmap_macro.rs

```diff
@@ -0,0 +1,27 @@
+// In lib.rs create a public hash_map! macro which allows to create HashMap like this:
+// let map = hash_map!(
+// 42 => true, 64 => false, 128 => true,
+// );
+// Write an integration test fo the macro
+
+#[macro_export]
+// Source: src/macros.rs - vec!
+macro_rules! hashmap {
+    () => {
+        HashMap::<u64, bool>::new();
+    };
+    ($($key:expr => $value:expr $(,)?)+) => {
+        HashMap::<u64, bool>::from([$(($key, $value) ,)*]);
+    };
+}
+
+#[macro_export]
+// Source: src/macros.rs - vec!
+macro_rules! hashmap_poorgeneric {
+    ( $(<)+ $key_type:ty $(,)+ $value_type:ty $(>)+)  => {
```

Comment:
You do not need to provide type annotation to this macro. If needed, user can provide type annotations themselves, e.g. `let x: HashMap<u64, u32> = hashmap!( 1=> 3, 2 => 4)`.

Reply:
done